synapse/cortex.py

```diff
@@ -193,38 +193,41 @@ def getCoreInfo(self):
         '''
         Adds a trigger to the cortex
         '''
-        iden = self.cell.triggers.add(self.user.name, condition, query, info=info)
+        iden = self.cell.triggers.add(self.user.iden, condition, query, info=info)
         return iden
 
-    def _auth_check(self, user):
+    def _trig_auth_check(self, useriden):
         ''' Check that, as a non-admin, may only manipulate resources created by you. '''
-        username = None if self.user is None else self.user.name
-        if username is not None and (not self.user.admin) and username != user:
+        if not self.user.admin and useriden != self.user.iden:
             raise s_exc.AuthDeny(user=self.user.name, mesg='As non-admin, may only manipulate triggers created by you')
 
     async def delTrigger(self, iden):
         '''
         Deletes a trigger from the cortex
         '''
         trig = self.cell.triggers.get(iden)
-        self._auth_check(trig.get('user'))
+        self._trig_auth_check(trig.get('useriden'))
         self.cell.triggers.delete(iden)
 
     async def updateTrigger(self, iden, query):
         '''
         Change an existing trigger's query
         '''
         trig = self.cell.triggers.get(iden)
-        self._auth_check(trig.get('user'))
+        self._trig_auth_check(trig.get('useriden'))
         self.cell.triggers.mod(iden, query)
 
     async def listTriggers(self):
         '''
         Lists all the triggers that the current user is authorized to access
         '''
-        username = None if self.user is None else self.user.name
-        return [(iden, trig) for (iden, trig) in self.cell.triggers.list()
-                if username is None or self.user.admin or trig.get('user') == username]
+        trigs = []
+        for (iden, trig) in self.cell.triggers.list():
+            if not (self.user.admin or trig['useriden'] == self.user.iden):
+                continue
+            trig['username'] = self.cell.auth.user(trig['useriden']).name
```

Comment:
what happens if a user is deleted from the system that made a trigger? would this blow up and cause all triggers to be unlistable by a admin user?

Reply:
Good point. Fixed.